42201.md

```diff
@@ -0,0 +1,71 @@
+![image](ZenIP-42001/ZenIP-42000-1.png)
+
+# ZenIP-42001
+
+    ZenIP: 42001
+    Title: ZenIP Process
+    Owner: Paolo Tagliaferri, <paolotagliaferri@horizenlabs.io>; cronic, <cronic@horizenlabs.io>
+    Discussions-To: Paolo Tagliaferri, <paolotagliaferri@horizenlabs.io>
+    Comments-URI: https://horizen.global/invite/discord
+    Status: Draft 
+    Type: Process
+    Created: 2022-04-07
+    License: MIT
+
+## Table of Contents
+
+<!--ts-->
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Specification](#specification)
+- [Notes](#Notes)
+
+## Abstract
+
+This ZenIP describes the enhancements to Zen v3.1.0. This version is a mandatory upgrade and includes the following:
+* new fields to `getblocktemplate` RPC command
+* new API to retrieve all versions for specified sidechain IDs
+* new value pair added to extend the `getscgenesisinfo` RPC commmand
+
+
+## Motivation
+
+The enhancements extend functionality for block template requests and to identify both sidechain versions and IDs together. These enhancements have already been implemented with v3.1.0.
+
+## Specification
+
+**Block Template Request Extensions**
+
+The `getblocktemplate` method includes two additional return values, based on the optional parameter `roots`:
+ 
+| Key | Required | Type | Description |
+|---|---|---|---|
+| merkleTree | no  | string  | Merkleroot calculated on transactions and certificates |
+| scTxsCommitment | no | string | Calculated on certificates |
+
+*Note: If you construct `coinbasetxn` outside of zend, use `getblockmerkleroots` RPC to calculate these values.
+
+**Sidechain versioning**
+
+The `GET_SIDECHAINS_VERSIONS` websocket API returns sidechain versions for provided sidechain IDs.
+
+The Sidechain Creation Output now includes sidechain version information. The `sc_create`, `createrawtransaction`, and `getscinfo` RPC commands are updated to include the `version` argument. This field is now mandatory when creating a new sidechain. Also, the `version` field is stored in `CTxScCreationOut`, `withdrawalEpochLength`, and `ScFixedParamaters`. The `sidechainVersion` parameter is added to `GetFieldElement`.
```

Comment:
Just for better clarity, I would highlight that `withdrawalEpochLength` has been used to store one byte for the `version` field.
```suggestion
The Sidechain Creation Output now includes sidechain version information. The `sc_create`, `createrawtransaction`, and `getscinfo` RPC commands are updated to include the `version` argument. This field is now mandatory when creating a new sidechain. Also, the `version` field has been added to `CTxScCreationOut` and `ScFixedParamaters`, it is stored in the first byte of the `withdrawalEpochLength`. The `sidechainVersion` parameter is added to `GetFieldElement`.
```